src/spyglass/utils/dj_helper_fn.py

```diff
@@ -8,6 +8,32 @@
 from .nwb_helper_fn import get_nwb_file
 
 
+def _delete(self: dj.Table, **kwargs) -> None:
+    descendants = self.descendants()
+    rows = self.fetch()
+    parents = []
+    parts = []
+    for table in descendants:
+        parent = table.split(".")[-1].split("`__")[-1]
+        if len(parent.split("__")) > 1:
+            parts.append(table)
+        else:
+            parents.append(table)
+    parts_no_parents = [
+        table
+        for table in parts
+        if (table.replace(f'__{table.split("__")[-1]}', "") + "`")
+        not in parents
+    ]
+    for part in parts_no_parents:
+        table = f"{part.split('.')[0]}.{part.split('.')[1].split('__')[0]}`"
```

Comment:
Just code wise, you're doing the same operation twice here so it would probably make sense to do something more like:
```python
first_part, last_part = part.split('.')
last_part = last_part.split(('__')[0]
table = f"{first_part}.{last_part}`"
conn = dj.conn()
```
for readability